### DISCUSSION TITLE
How can we share the Link, redirect, usePathname, and useRouter functions across modules in a monorepo project?

### DISCUSSION THREAD

**Discussion by rogerogers** [amannn/next-intl#1559] [Q&A]
### Is your feature request related to a problem? Please describe.

My monorepo project is expected to have many modules. I want to share these several methods.

### Describe the solution you'd like

not yet

### Describe alternatives you've considered

use next/navigation

**Comment by amannn** [comment]
What's stopping you from calling `createNavigation` in a shared module?

Moving to a discussion as this is a usage question …

**Reply by rogerogers** [reply]
I tried again and it worked. I had some error messages when I tried yesterday. Thank you very much.

**Reply by pbortnick** [reply]
@rogerogers any tips on implementing this? 🙏

**Reply by rogerogers** [reply]
> @rogerogers any tips on implementing this? 🙏

I created a separate project to share these methods of mine

```js
import { createNavigation } from 'next-intl/navigation';
import { defineRouting } from 'next-intl/routing';

// Can be imported from a shared config
export const locales = ['en', 'zh'] as const;

export type Locale = (typeof locales)[number];

export const routing = defineRouting({
  // A list of all locales that are supported
  locales: locales,

  // Used when no locale matches
  defaultLocale: locales[0],
});

// Lightweight wrappers around Next.js' navigation APIs
// that will consider the routing configuration
export const { Link, redirect, usePathname, useRouter, getPathname } =
  createNavigation(routing);

```
